### PR TITLE
allow Navbar to have zero or one children

### LINKED ISSUE #524: Navbar component breaks when enclosing only one direct child component.
**Describe the bug**
Navbar component breaks when enclosing only one direct child component.

**To Reproduce**
Include a Navbar containing one child component. Eg. Copy the Navbar story code and remove Brand and Links in order to leave only one direct child component : 

` <Navbar color='white'>
        <Link color='white' href={'/docs/'}>
          Docs
        </Link>
  </Navbar>`

**Expected behavior**
It should render succesfully.

**Screenshots**
Navbar is not rendered at all and console prints the following error:
image

**Specifications (please complete the following information):**
  - Browser: Chrome 67  whatsmybrowser.org/b/52G5VWD
  - OS: Ubuntu 16.04 LTS
  - Environment: development (webpack-dev-server --mode development)

**Additional context**
When added another direct child component Navbar renders normally:

` <Navbar color="white">
      <Link color="white" href="/docs/">
        Docs
      </Link>
      <Link color="white" href="/changelog/">
        changelog
      </Link>
  </Navbar>`

I ran into this error when trying to enclose Navbar child components inside the recently documented Flex container.

### PR DESCRIPTION
**Pull request by mcchrish**
Fixes #524

### CODE REVIEW

File: __tests__/components/Navbar.spec.js [hunk 1]
```diff
@@ -1,26 +1,52 @@
...
-        <Link color='white' href={'/gitter/'}>
-          gitter
-        </Link>
-      </Navbar>
-    </Provider>
-  )
-  let tree = component.toJSON()
-  expect(tree).toMatchSnapshot()
+describe('Navbar renders correctly', () => {
+  test('with more than one children props', () => {
```
Comment (LucianBuzzo) [inline __tests__/components/Navbar.spec.js]: Nit: change `props` to `prop`

File: __tests__/components/Navbar.spec.js [hunk 2]
```diff
@@ -1,26 +1,52 @@
...
+            gitter
+          </Link>
+        </Navbar>
+      </Provider>
+    )
+    let tree = component.toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+
+  test('with a single children props', () => {
```
Comment (LucianBuzzo) [inline __tests__/components/Navbar.spec.js]: Nit: change `props` to `prop`

File: __tests__/components/Navbar.spec.js [hunk 3]
```diff
@@ -1,26 +1,52 @@
...
+            Docs
+          </Link>
+        </Navbar>
+      </Provider>
+    )
+    let tree = component.toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+
+  test('with no children props', () => {
```
Comment (LucianBuzzo) [inline __tests__/components/Navbar.spec.js]: Nit: change `props` to `prop`

**Comment by mcchrish** [comment]
@LucianBuzzo I've updated the commit and test descriptions.

**Pull request merged**